### PR TITLE
Fix(pubsub): Upgrade Cloud Functions Pub/Sub to Java25

### PR DESCRIPTION
**Pull request by Kef131**
## Description
Fixes # 
b/470272226
Upgrades the Cloud Functions Pub/Sub Publish Message sample to Java 25:
- Sets `maven.compiler.source` and `maven.compiler.target` to `25`.
- Upgrades `shared-configuration` parent version to `1.2.2`.
- Overrides `jacoco-maven-plugin` to `0.8.14` to support Java 25 (Class file major version 69). - Configures `jacoco-maven-plugin` executions (`prepare-agent` and `report`) to enable dynamic bytecode instrumentation for test coverage collection and report generation during the test lifecycle as mentioned on bug.

## Checklist
- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration` (1.2.2)
- [x] Appropriate changes to README are included in PR (None needed)
- [ ] These samples need a new **API enabled** in testing projects to pass (None)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (None)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only** (Note: ErrorProne 2.18.0 and Spotbugs 4.7.3.3 from parent pom have compatibility warnings/errors under Java 25)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: functions/pubsub/publish-message/pom.xml [hunk 1]
```diff
@@ -27,12 +27,13 @@
...
-    <version>1.2.0</version>
+    <version>1.2.2</version>
   </parent>
 
   <properties>
-    <maven.compiler.target>11</maven.compiler.target>
-    <maven.compiler.source>11</maven.compiler.source>
+    <!-- Change this to 25 in your workspace -->
+    <maven.compiler.target>21</maven.compiler.target>
+    <maven.compiler.source>21</maven.compiler.source>
```
[the remark this answers is not in the document]
Comment (Kef131) [reply functions/pubsub/publish-message/pom.xml:36]: Here we have an issue with Kokoro when upgrading samples to Java25
Kokoro fails on both Java 11 and Java 17 presubmit builds, caused by the same issue: the repository's test runner ([btlr](https://github.com/googleapis/btlr) ) does not yet support Java version 25 as a compiler target in pom.xml, and fails immediately when encountering it.

This PR follows the workaround applied in [appengine-java25/helloworld/pom.xml](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/2c245f768270259d9e6d62d876461bf858085ed4/appengine-java25/helloworld/pom.xml#L33) and [flexible/java-25/websocket-jetty/pom.xml](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/2c245f768270259d9e6d62d876461bf858085ed4/flexible/java-25/websocket-jetty/pom.xml#L37) ) which is using Java 21 Target in POM

We would set maven.compiler.target and maven.compiler.source to 21 in pom.xml, but keep the build/runtime environment in the Dockerfile configured to use Java 25.

**Review by angelcaamal — APPROVED** [review]
LGTM!

**Comment by XrossFox** [comment]
LGTM!

**Review by iennae — APPROVED** [review]
